File: refactoring_test/rscore/tests/application/EncapsulateAllPublicFieldsTest.scala
package rscore.tests.application
import org.junit.Test
import rscore.dsl.entity.collection.By
import rscore.dsl.entity.collection.With

class EncapsulateAllPublicFieldsTest extends ApplicationTest{
	@Test
	def publicフィールドのうちstaticでないものをすべてカプセル化する(): Unit = {
		val testName = "EncapsulateAllPublicFieldsTest"
		prepareTest(testName)

		val c = getCurrentProject().pkg(this.testGroupIdentifier).classes().Select_one(By name testName)
		val pfs = c.fields.select(By modifier "public").select(By modifier (With out "static"))
		pfs.foreach(_.encapsulate())

		doAssert(testName)
	}
}

File: test_resources_input/application/EncapsulateAllPublicFieldsTest.java
package application;

public class EncapsulateAllPublicFieldsTest {
	public int a;
	public String b;
	
	private int p;
	private String q;
	
	protected int x;
	protected String y;
	
	public static int aa;
	private static final int bb = 100;
	
	void m(){
		System.out.println(a + b + p + q + x + y + aa + bb);
	}
}

class Callee{
	void m(){
		EncapsulateAllPublicFieldsTest e = new EncapsulateAllPublicFieldsTest();
		System.out.println(e.a);
		System.out.println(e.b);
		e.a = 100;
		e.b = "string";
		
		System.out.println(EncapsulateAllPublicFieldsTest.aa);
	}
}
